packages/strapi/lib/core-api/service.js

```diff
@@ -1,16 +1,18 @@
 'use strict';
 
+const _ = require('lodash');
 /**
  * default service
  *
  */
 
-module.exports = ({ model, strapi }) => {
-  if (model.kind === 'singleType') {
-    return createSingleTypeService({ model, strapi });
-  }
+const NON_EDITABLE_FIELDS = ['created_by', 'updated_by', 'published_at'];
+const sanitizeInput = (data, model) => {
+  const timestamps = _.get(model, ['options.timestamps'], []);
+  const excludedFields = NON_EDITABLE_FIELDS.concat(timestamps);
+  const allowedFields = _.difference(Object.keys(model.attributes), excludedFields);
```

Comment:
Keep extra fields the user might want to do things with. This allows more flexibility for customization in the lifecycles